pkg/ruler/ruler.go

```diff
@@ -444,3 +453,130 @@ func (r *Ruler) newManager(ctx context.Context, userID string) (*promRules.Manag
 	}
 	return promRules.NewManager(opts), nil
 }
+
+// GetRules retrieves the running rules from this ruler and all running rulers in the ring if
+// sharding is enabled
+func (r *Ruler) GetRules(ctx context.Context, userID string) ([]*rules.RuleGroupDesc, error) {
+	if r.cfg.EnableSharding {
+		return r.getShardedRules(ctx, userID)
+	}
+
+	return r.getLocalRules(userID)
+}
+
+func (r *Ruler) getLocalRules(userID string) ([]*rules.RuleGroupDesc, error) {
+	groupDescs := []*rules.RuleGroupDesc{}
+
+	var groups []*promRules.Group
+	r.userManagerMtx.Lock()
+	mngr, exists := r.userManagers[userID]
+	if exists {
+		groups = mngr.RuleGroups()
+	}
+	r.userManagerMtx.Unlock()
+
+	for _, group := range groups {
+		interval := group.Interval()
+		groupDesc := &rules.RuleGroupDesc{
+			Name:      group.Name(),
+			Namespace: group.File(),
```

Comment:
Two things:
1. Why is it called `Namespace` if it then maps the `File` in the API response (see `api.go`)?
2. Is it really a good thing exposing the path of the file inside Cortex? Looks we may disclose a sensitive information in a Cortex as a service scenario

Reply:
2. I updated the downstream path to strip the internal cortex file path. The only thing that will be left is the namespace of the users rule group.

The reason Namespace was chosen instead of file is basically to avoid have file be an API object when interacting with Cortex.